File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> A, vector<int> B) {
    int answer = 0;
    sort(A.rbegin(), A.rend());
    sort(B.rbegin(), B.rend());
    
    int Aidx = 0;
    int Bidx = 0;
    
    while(Aidx < A.size())
    {
        if (B[Bidx] > A[Aidx])
        {
            ++answer;
            ++Bidx;
        }
       
        ++Aidx;
    }
    
    return answer;
}

File: 프로그래머스/3/12987. 숫자 게임/숫자 게임.java
import java.util.Arrays;
import java.util.Collections;

public class Solution {

    public static int solution(int[] A, int[] B) {
        int answer = 0;

        Integer[] boxedA = Arrays.stream(A).boxed().sorted(Collections.reverseOrder()).toArray(Integer[]::new);
        Integer[] boxedB = Arrays.stream(B).boxed().sorted(Collections.reverseOrder()).toArray(Integer[]::new);

        int aIdx = 0;
        int bIdx = 0;

        while (aIdx < boxedA.length) {
            if (boxedB[bIdx] > boxedA[aIdx]) {
                answer++;
                bIdx++;
            }

            aIdx++;
        }

        return answer;
    }
}